lang/en/typeshed/stdlib/microbit/microphone.pyi

```diff
@@ -53,18 +54,107 @@ def set_threshold(event: SoundEvent, value: int) -> None:
 
     Example: ``microphone.set_threshold(SoundEvent.LOUD, 250)``
 
-    A high threshold means the event will only trigger if the sound is very loud (>= 250 in the example).
+    The ``SoundEvent.LOUD`` event is triggered when the sound level crosses the 
+    threshold from "quiet" to "loud", and the ``SoundEvent.QUIET`` event is 
+    triggered when the sound level crosses from "loud" to "quiet".
+
+    If the ``SoundEvent.LOUD`` threshold is set lower than the 
+    ``SoundEvent.QUIET`` threshold, then the ``SoundEvent.QUIET`` threshold 
+    will decrease by one unit below the ``SoundEvent.LOUD`` threshold. If the 
+    ``SoundEvent.QUIET`` threshold is set higher than the ``SoundEvent.LOUD``
+    threshold, then the ``SoundEvent.LOUD`` threshold will increase by one unit 
+    above the ``SoundEvent.QUIET`` threshold.
 
     :param event: A sound event, such as ``SoundEvent.LOUD`` or ``SoundEvent.QUIET``.
-    :param value: The threshold level in the range 0-255.
+    :param value: The threshold level in the range 0-255. Values outside this range will be clamped.
     """
     ...
 
 def sound_level() -> int:
-    """Get the sound pressure level.
+    """Get the sound pressure level in the range 0 to 255.
 
     Example: ``microphone.sound_level()``
 
     :return: A representation of the sound pressure level in the range 0 to 255.
     """
     ...
+
+def sound_level_db() -> int:
+    """Get the sound pressure level in decibels.
+
+    Example: ``microphone.sound_level_db()``
+
+    :return: A representation of the sound pressure level in decibels (dB).
+    """
+    ...
```

Comment:
May be subject to change following https://github.com/bbcmicrobit/micropython/pull/803 PR